### ISSUE TITLE
9$: False positive deleting global variable

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#434]
In Node.js, the following code is legal and the `delete` has useful side effects:
```js
global.banana = 42;
console.log(delete banana);
console.log(global.banana);
```
However, quick-lint-js falsely reports a warning saying that the `delete` is pointless:
```js
hello.js:2:13: warning: redundant delete statement on variable [E086]
hello.js:2:20: warning: use of undeclared variable: banana [E057]
```

E086 should only apply to local variables and module variables (and maybe non-configurable globals).

**Comment by strager** [comment]
cc @singalhimanshu who added this diagnostic in commit 8ab6bed96f16678b686395643728e580e6e8d0e0.

**Comment by strager** [comment]
Fixed in commit 68bddc371c69727b4d4cb4482f351ddf564d02d6.
